Net/SMTP.php

```diff
@@ -198,6 +202,7 @@ public function __construct($host = null, $port = null, $localhost = null,
         /* These standard authentication methods are always available. */
         $this->setAuthMethod('LOGIN', array($this, 'authLogin'), false);
         $this->setAuthMethod('PLAIN', array($this, 'authPlain'), false);
+        $this->setAuthMethod('GSSAPI', array($this, 'authGSSAPI'), false);
```

Comment:
I would register the method first, and do it only if principal and cname is set. Maybe checking for krb5 extension should be done here (as we do with checking SASL class).